File: src/main/java/com/refinedmods/refinedstorage/screen/grid/filtering/NameGridFilter.java
package com.refinedmods.refinedstorage.screen.grid.filtering;

import com.refinedmods.refinedstorage.screen.grid.stack.IGridStack;

import java.util.function.Predicate;

public class NameGridFilter implements Predicate<IGridStack> {
    private String name;

    public NameGridFilter(String name) {
        this.name = name.toLowerCase();
    }

    @Override
    public boolean test(IGridStack stack) {
        return stack.getName().toLowerCase().contains(name);
    }
}

File: src/main/kotlin/com/refinedmods/refinedstorage/screen/grid/filtering/NameGridFilter.kt
package com.refinedmods.refinedstorage.screen.grid.filtering

import com.refinedmods.refinedstorage.screen.grid.stack.IGridStack
import java.util.function.Predicate

class NameGridFilter(name: String) : Predicate<IGridStack> {
    private val name: String
    override fun test(stack: IGridStack): Boolean {
        return stack.name.toLowerCase().contains(name)
    }

    init {
        this.name = name.toLowerCase()
    }
}